### PR TITLE
Autocomplete component

### LINKED ISSUE #111: Autocomplete component
image

### PR DESCRIPTION
**Pull request by 9teen90nine**
## PR includes

- [x] Feature

## Related issues

Resolve #111 

## QA Instructions, Screenshots, Recordings
  
Added option to add a search into a dropdown. Also added a story for FilterDropdown displaying the feature.
image

### CODE REVIEW

File: src/components/Dropdown.tsx [hunk 1]
```diff
@@ -27,12 +29,13 @@ export interface DropdownItemProps {
...
 }
 
-interface DropdownProps {
+export interface DropdownProps {
     renderItem: (props: DropdownItemProps) => React.ReactNode;
     renderTrigger: (props: DropdownTriggerProps) => React.ReactNode;
     text?: string;
     value?: any;
     items?: any[];
+    searchFilter?: (item: any, query: string) => boolean;
```
Comment (awinogradov) [inline src/components/Dropdown.tsx:38]: `searchFilter` not correct :( `searchQuery `or `query` would be better

File: src/components/Dropdown.tsx [hunk 2]
```diff
@@ -8,6 +8,8 @@ import { useKeyPress } from '../hooks/useKeyPress';
 import { useKeyboard, KeyCode } from '../hooks/useKeyboard';
 
 import { Popup } from './Popup';
+import Input from './Input';
```
Comment (awinogradov) [inline src/components/Dropdown.tsx:11]: We must delete default exports at all from package

Comment (9teen90nine) [reply src/components/Dropdown.tsx:11]: I deleted default export of Dropdown, but I believe it should be another issue - #223

File: src/components/Dropdown.tsx (hunk 1 above)
Comment (9teen90nine) [reply src/components/Dropdown.tsx]: changed to `onSearchChange`

File: src/components/Dropdown.tsx [hunk 3]
```diff
@@ -177,12 +192,32 @@ export const Dropdown = React.forwardRef<HTMLDivElement, DropdownProps>(
...
+                                placeholder="search"
+                                value={query}
+                                ref={searchBoxRef}
+                                onChange={({ target }) => {
+                                    const { value = '' } = target || {};
+                                    setQuery(value);
+                                    onSearchChange && onSearchChange(value);
+                                }}
+                                iconRight={
+                                    <CrossIcon
```
Comment (awinogradov) [inline src/components/Dropdown.tsx:206]: Show if query exists

**Review by awinogradov — APPROVED** [review]

**Pull request merged**